src/Symfony/Component/HttpKernel/Bundle/SimpleBundle.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Bundle;
+
+use Symfony\Component\Config\Definition\Builder\NodeParentInterface;
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\HttpKernel\DependencyInjection\Extension;
+
+/**
+ * A all-in-one implementation of {@link BundleInterface},
+ * {@link ExtensionInterface} and {@link ConfigurationInterface}.
+ *
+ * @author Guilhem N. <egetick@gmail.com>
+ */
+abstract class SimpleBundle extends Bundle implements ConfigurationInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    final public function getContainerExtension()
+    {
+        return new SimpleExtension($this);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    final public function getConfigTreeBuilder()
+    {
+        $treeBuilder = new TreeBuilder();
+        $rootNode = $treeBuilder->root($this->getAlias());
+
+        $this->buildConfiguration($rootNode);
+
+        return $treeBuilder;
+    }
+
+    protected function buildConfiguration(NodeParentInterface $rootNode)
+    {
+    }
+
+    protected function buildContainer(array $config, ContainerBuilder $container)
+    {
+    }
+}
+
+/**
+ * Simple {@link ExtensionInterface} supporting {@link MicroBundle}.
+ *
+ * @internal
+ */
+final class SimpleExtension extends Extension
+{
+    private $bundle;
+    private $buildContainer;
+
+    public function __construct(SimpleBundle $bundle, Closure $buildContainer)
+    {
+        $this->bundle = $bundle;
+        $this->buildContainer = Closure::bind(function (array $config, ContainerBuilder $container) {
```

Comment:
This is usually called only once during building phase right? ie. do we really need property cache? 2nd. constructor arg is unused now ;-)